includes/multi-currency/Compatibility/WooCommerceSubscriptions.php

```diff
@@ -158,13 +159,15 @@ public function override_selected_currency( $return ) {
 
 		$switch_cart_items = $this->get_subscription_switch_cart_items();
 		if ( 0 < count( $switch_cart_items ) ) {
-			$switch_cart_item = array_shift( $switch_cart_items );
-			return get_post_meta( $switch_cart_item['subscription_switch']['subscription_id'], '_order_currency', true );
+			$switch_cart_item    = array_shift( $switch_cart_items );
+			$switch_subscription = $this->get_subscription( $switch_cart_item['subscription_switch']['subscription_id'] );
+			return $switch_subscription ? $switch_subscription->get_currency() : $return;
 		}
 
 		$subscription_resubscribe = $this->cart_contains_resubscribe();
 		if ( $subscription_resubscribe ) {
-			return get_post_meta( $subscription_resubscribe['subscription_resubscribe']['subscription_id'], '_order_currency', true );
+			$switch_subscription = $this->get_subscription( $subscription_resubscribe['subscription_resubscribe']['subscription_id'] );
+			return $switch_subscription ? $switch_subscription->get_currency() : $return;
```

Comment:
Nitpick: The `$switch_subscription` variable name could be confusing since it is referring to a `resubscribe` rather than a `switch` here.

Reply:
Good spotting, that would annoy me as it's misleading! Fixed here: a5ccf31500cf6f27c3b21ed25ca2a120ed729b84